pkg/api/controller.go

```diff
@@ -3026,7 +3026,12 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 }
 
 func (c *Controller) sendResetPasswordEmail(email string, token string) error {
-	return c.Emailer.SendEmail([]string{email}, token, token, nil)
+	if c.Emailer.Limiter.Allow() {
+		return c.Emailer.SendEmail([]string{email}, token, token, nil)
+	}
+	err := ErrRateLimitExceeded
+	c.Logger.WithError(err).WithField("email", email).Debug("Email sending rate limit was exceeded")
+	return err
```

Comment:
This code should go inside `c.Emailer.SendEmail()` (or `SendEmailWithLimit`), not inside the Controller.  The general rule is that you should prefer not to dig inside another object's internals.

Reply:
Went with SendEmailWithLimit, to give the ability to send email without limits if the caller is intrested